### ISSUE TITLE
Tests with a comma in their name cannot be run from command line

### ISSUE THREAD

**Issue by cdmh** [catchorg/Catch2#932]
## Description
I have lots of tests and one has a comma in its name "Fill buffer with random PTZ images and keep wrapping, then empty". If I specify this on the command line, I get "No tests ran". If I rename it to "Fill buffer with random PTZ images and keep wrapping then empty" (i.e. remove the command), then the test runs as expected. The behaviour is the same in MSVS and from a command prompt.

### Extra information
* Catch version: **v1.8.2**
* Operating System: **Windows 10**
* Compiler+version: **MSVS 2017 Professional**

**Comment by philsquared** [comment]
This is due to the old problem of macro arguments with commas being interpreted as multiple arguments.
This has been fixed in Catch2 (which will go live very soon) as it would have relied on multiplicative duplication in Catch1.x due to not being able to rely on variadic macros.

**Comment by kirbyfan64** [comment]
@philsquared 

> This has been fixed in Catch2 (**which will go live very soon**)

Any estimates? ;)

**Comment by philsquared** [comment]
Sorry, @cdmh - on reviewing this issue I realise I misread your original problem!
It's the _test_ name that has a comma. I was thinking of commas in assertions.

In fact, to run your test case with a comma in it you just need to escape the comma (with a backslash).
So, in your example, you'd run with:

`"Fill buffer with random PTZ images and keep wrapping\, then empty"`

This is a bit of a pain, and you might still want to avoid using commas in test names for that reason - but if you do use them that's how you do it.
This is to do with how test specs are parsed (as you can combine test case names, patterns and tags - and comma means "tests matching this, and that")

**Comment by philsquared** [comment]
@kirbyfan64 - very soon :-)

**Comment by maxim-banaev** [comment]
It seems that Catch v2.9.2 has regression because I cannot run test with comma in name.

**Comment by horenmar** [comment]
Have you tried current version?

**Comment by maxim-banaev** [comment]
Confirm that it's fine with current (2.13.0) version. Thanks.

**Comment by pkrukp** [comment]
Doesn't work with v3.3.2:

```
$ ./tests.exe "Operation isLeftShift, isRightShift, etc"
Filters: "Operation isLeftShift","isRightShift","etc"
Randomness seeded to: 1483106191
No test cases matched '"Operation isLeftShift"'
No test cases matched '"isRightShift"'
No test cases matched '"etc"'
===============================================================================
No tests ran
```

**Comment by horenmar** [comment]
Where is the backslash escape?

**Comment by pkrukp** [comment]
There is no backlash - not needed?

> This has been fixed in Catch2 (which will go live very soon)

**Comment by horenmar** [comment]
And the follow up

> Sorry, @cdmh - on reviewing this issue I realise I misread your original problem!
It's the test name that has a comma. I was thinking of commas in assertions.

Once upon a time, there was issue with the assertion macros (`REQUIRE`, `CHECK`, etc) being non-variadic, thus breaking if there was a comma in the expression inside (due to how preprocessor splits arguments on commas).

The CLI always needed escaping, because `,` is the OR operator, so `[abc],[def]` means "tests tagged with [abc] or [def]", and similarly `"foo bar","baz baz"` means "test(s) named 'foo bar' or 'baz baz'".

**Comment by pkrukp** [comment]
Hmm but I didn't use `"foo bar","baz baz"`. I agree, if I used it, it make sense to treat it as two names split by comma.

I used `"foo bar, baz baz"`. I guess shell stripped quotes... Not sure if that changes anything for you.

So maybe it would be helpful for new users to warn `we didn't find any test cases matching your patterns, but found a test case matching pattern with escaped commas like this: "foo bar\, baz baz"` ?
